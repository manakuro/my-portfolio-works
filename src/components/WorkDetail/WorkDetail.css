/* ------------------------------
  Overlay scale animation
 --------------------------------*/
.work-detail-Overlay {
  width: 50%;
  height: 100%;
  position: fixed;
  left: 50%;
  top: 0;
  z-index: 3001;
  transform: translateX(100%);
  background: #fff;
  padding: 0 30px 30px 30px;
  overflow-y: scroll; }

.work-detail-Circle {
  border-radius: 50%;
  background: #f4f7f6;
  width: 100px;
  height: 100px;
  position: absolute;
  left: -1000px;
  top: 0;
  z-index: 2001;
  opacity: 0; }

.scale-enter-active, .scale-appear-active {
  opacity: 1;
  transform: scale(40);
  transition: transform 2s cubic-bezier(0.215, 0.61, 0.355, 1) 0s; }

.scale-enter-done, .scale-exit {
  opacity: 1;
  transform: scale(40); }

.scale-exit-active {
  opacity: 0;
  transition: opacity .5s; }

/* ------------------------------
  Work Content
 --------------------------------*/
.work-detail-Content_Wrapper {
  max-width: 760px;
  margin: 0 auto; }

.work-detail-Content_Heading {
  margin-top: 80px;
  margin-bottom: 80px;
  font-size: 4rem;
  line-height: 1.5; }

.work-detail-ExpandOverlay {
  position: absolute;
  padding: 20px;
  right: 0;
  top: 0;
  font-size: 1.2rem;
  cursor: pointer; }

.slide-works-enter-active {
  opacity: 1;
  transform: translateX(0);
  transition: transform 0.5s cubic-bezier(0.215, 0.61, 0.355, 1) 0.4s; }

.slide-works-enter-done, .slide-works-exit {
  opacity: 1;
  transform: translateX(0); }

.slide-works-exit-active {
  opacity: 0;
  transform: translateX(100%);
  transition: transform 0.7s cubic-bezier(0.215, 0.61, 0.355, 1), opacity 0.7s; }

.expand-works-enter-active {
  width: 100%;
  left: 0;
  transform: translateX(0);
  transition: 0.5s cubic-bezier(0.215, 0.61, 0.355, 1); }

.expand-works-enter-done, .expand-works-exit {
  width: 100%;
  left: 0;
  transform: translateX(0); }

.expand-works-exit-active {
  width: 50%;
  left: 50%;
  transform: translateX(0);
  transition: 0.5s cubic-bezier(0.215, 0.61, 0.355, 1); }

.work-detail-BackButton {
  position: fixed;
  top: 25px;
  left: 25px;
  z-index: 3001;
  transform: translateX(-100px); }
  .work-detail-BackButton .icon {
    font-size: 2rem;
    cursor: pointer;
    width: 60px;
    height: 60px;
    background: #fff;
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative; }
    .work-detail-BackButton .icon.expanded {
      background: #f4f7f6;
      transition: background .25s; }

.slide-button-enter-active {
  opacity: 1;
  transform: translateX(0);
  transition: transform 0.5s cubic-bezier(0.215, 0.61, 0.355, 1) 0.4s; }

.slide-button-enter-done, .slide-button-exit {
  opacity: 1;
  transform: translateX(0); }

.slide-button-exit-active {
  opacity: 1;
  transform: translateX(-100px);
  transition: transform 0.7s cubic-bezier(0.215, 0.61, 0.355, 1), opacity 0.7s; }

/* ------------------------------
  Works content left
 --------------------------------*/
.work-detail-ContentLeftWrapper {
  position: relative;
  z-index: 3000;
  opacity: 0; }

.work-detail-ContentLeft {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 3000;
  background: transparent;
  padding: 0 50px; }
  .work-detail-ContentLeft_Inner {
    width: 90%; }

/* ------------------------------
  Uncover
 --------------------------------*/
.work-detail-Uncover {
  overflow: hidden;
  background-image: none !important;
  position: relative; }
  .work-detail-Uncover_Slices {
    flex-direction: column;
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    display: flex; }
  .work-detail-Uncover_Slice {
    flex: 1;
    color: #f4f7f6;
    background: currentColor; }

/* ------------------------------
  Overwrite github-markdown-css
 --------------------------------*/
.markdown-body {
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  color: #333; }
  .markdown-body ul, .markdown-body li {
    list-style: initial; }
