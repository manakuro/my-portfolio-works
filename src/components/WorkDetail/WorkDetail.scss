$bgOverlay: rgb(244, 247, 246);

/* ------------------------------
  Overlay scale animation
 --------------------------------*/
.work-detail-Overlay {
  width: 50%;
  height: 100%;
  position: fixed;
  left: 50%;
  top: 0;
  z-index: 3001;
  transform: translateX(100%);
  background: #fff;
  padding: 0 30px 30px 30px;
  overflow-y: scroll;
}

.work-detail-Circle {
  border-radius: 50%;
  background: rgb(244, 247, 246);
  width: 100px;
  height: 100px;
  position: absolute;
  left: -1000px;
  top: 0;
  z-index: 2001;
  opacity: 0;
}

.scale-enter-active, .scale-appear-active {
  opacity: 1;
  transform: scale(40);
  transition: transform 2s cubic-bezier(0.215, 0.610, 0.355, 1.000) 0s; // easeOutCubic
}

.scale-enter-done, .scale-exit {
  opacity: 1;
  transform: scale(40);
}

.scale-exit-active {
  opacity: 0;
  transition: opacity .5s;
}

/* ------------------------------
  Work Content
 --------------------------------*/
.work-detail-Content_Wrapper {
  max-width: 760px;
  margin: 0 auto;
}

.work-detail-Content_Heading {
  margin-top: 80px;
  margin-bottom: 80px;
  font-size: 4rem;
  line-height: 1.5;
}

.work-detail-ExpandOverlay {
  position: absolute;
  padding: 20px;
  right: 0;
  top: 0;
  font-size: 1.2rem;
  cursor: pointer;
}

// enter transition
.slide-works-enter-active {
  opacity: 1;
  transform: translateX(0);
  transition: transform .5s cubic-bezier(0.215, 0.610, 0.355, 1.000) .4s;
}
.slide-works-enter-done, .slide-works-exit {
  opacity: 1;
  transform: translateX(0);
}

// exit transition
.slide-works-exit-active {
  opacity: 0;
  transform: translateX(100%);
  transition: transform .7s cubic-bezier(0.215, 0.610, 0.355, 1.000), opacity .7s;
}

// expand transition
.expand-works-enter-active {
  width: 100%;
  left: 0;
  transform: translateX(0);
  transition: .5s cubic-bezier(0.215, 0.610, 0.355, 1.000);
}

.expand-works-enter-done, .expand-works-exit {
  width: 100%;
  left: 0;
  transform: translateX(0);
}

.expand-works-exit-active {
  width: 50%;
  left: 50%;
  transform: translateX(0);
  transition: .5s cubic-bezier(0.215, 0.610, 0.355, 1.000);
}

.work-detail-BackButton {
  position: fixed;
  top: 25px;
  left: 25px;
  z-index: 3001;
  transform: translateX(-100px);

  .icon {
    font-size: 2rem;
    cursor: pointer;
    width: 60px;
    height: 60px;
    background: #fff;
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;

    &.expanded {
      background: #f4f7f6;
      transition: background .25s;
    }
  }
}

.slide-button-enter-active {
  opacity: 1;
  transform: translateX(0);
  transition: transform .5s cubic-bezier(0.215, 0.610, 0.355, 1.000) .4s;
}

.slide-button-enter-done, .slide-button-exit {
  opacity: 1;
  transform: translateX(0);
}

.slide-button-exit-active {
  opacity: 1;
  transform: translateX(-100px);
  transition: transform .7s cubic-bezier(0.215, 0.610, 0.355, 1.000), opacity .7s;
}



/* ------------------------------
  Works content left
 --------------------------------*/
.work-detail-ContentLeftWrapper {
  position: relative;
  z-index: 3000;
  opacity: 0;
}

.work-detail-ContentLeft {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 50%;
  height: 100%;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 3000;
  background: transparent;
  padding: 0 50px;

  &_Inner {
    width: 90%;
  }
}


/* ------------------------------
  Uncover
 --------------------------------*/
.work-detail-Uncover {
  overflow: hidden;
  background-image: none !important;
  position: relative;

  &_Slices {
    flex-direction: column;
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    display: flex;
  }

  &_Slice {
    flex: 1;
    color: $bgOverlay;
    background: currentColor;
  }
}


/* ------------------------------
  Overwrite github-markdown-css
 --------------------------------*/
.markdown-body {
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  color: #333;

  ul, li {
    list-style: initial;
  }
}



