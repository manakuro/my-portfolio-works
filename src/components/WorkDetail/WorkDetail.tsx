import * as React from 'react'
import { Animated } from 'react-animated-css'
import Anime, { AnimeProps } from 'react-anime'
import { CSSTransition } from 'react-transition-group'
import compose from 'recompose/compose'

// others
import './WorkDetail.css'
import addClassHtml from '@/utils/addClassHtml'
import removeClassHtml from '@/utils/removeClassHtml'
import WorkDetailContent from '@/components/WorkDetailContent/WorkDetailContent'
import * as classnames from 'classnames'
import { HomeState } from '@/modules/home/reducer'
import { HomeDispatchFromProps, HomeOwnProps } from '@/components/Home/Home'
import { withHandlers } from 'recompose'

export interface WorkDetailProps {
  toggleOverlay: HomeDispatchFromProps['toggleOverlay']
  toggleWorksContent: HomeDispatchFromProps['toggleWorksContent']
  toggleWorksContentAnimation: HomeDispatchFromProps['toggleWorksContentAnimation']
  toggleWorksContentExpand: HomeDispatchFromProps['toggleWorksContentExpand']
  updateCircle: HomeDispatchFromProps['updateCircle']
  updateWorkContentImg: HomeDispatchFromProps['updateWorkContentImg']
  fetchWork: HomeDispatchFromProps['fetchWork']

  workContentImg: HomeState['workContentImg']
  isShowOverlay: HomeState['isShowOverlay']
  isShowWorksContent: HomeState['isShowWorksContent']
  isExpandWorksContent: HomeState['isExpandWorksContent']
  circleStyle: HomeState['circleStyle']
  isShowWorksContentAnimation: HomeState['isShowWorksContentAnimation']
  targetWork: HomeState['targetWork']
  history: HomeOwnProps['history']
  match: HomeOwnProps['match']
}

const ANIME_PROPS: AnimeProps = {
  children: [],
  easing: 'easeInOutCirc',
  duration: 500,
  delay: (_, i: number, t: number) => Math.abs(t / 2 - i) * 80,
  translateX: '-100%',
  scale: 1,
  value: '',
  translateY: 0,
  rotate: 0,
  opacity: 1,
  color: '',
  backgroundColor: '',
  points: '',
  strokeDashoffset: 0,
}

export function renderWorkContentImgAnimation(
  props: WorkDetailProps,
): JSX.Element | null {
  return props.workContentImg !== '' &&
    props.isShowWorksContent &&
    !props.isExpandWorksContent ? (
    <Anime {...ANIME_PROPS} key={`anime-${props.workContentImg}`}>
      <div className="work-detail-Uncover_Slice" />
      <div className="work-detail-Uncover_Slice" />
      <div className="work-detail-Uncover_Slice" />
      <div className="work-detail-Uncover_Slice" />
      <div className="work-detail-Uncover_Slice" />
      <div className="work-detail-Uncover_Slice" />
    </Anime>
  ) : null
}

export function goBack(props: WorkDetailProps): void {
  props.history.goBack()
}

export function hideOverlay(props: WorkDetailProps): void {
  props.toggleWorksContent(false)
  removeClassHtml()

  props.toggleOverlay(false)
  props.toggleWorksContentAnimation(false)
  props.toggleWorksContentExpand(false)
}

export function onExitedOverlay(props: WorkDetailProps): void {
  props.updateCircle({
    top: '0px',
    left: '-1000px',
  })

  goBack(props)
}

export function showWorksContent(props: WorkDetailProps): void {
  props.toggleWorksContent(true)
  addClassHtml('hidden')
}

export function onEnteredShowWorksContent(props: WorkDetailProps): void {
  props.toggleWorksContentAnimation(true)
}

export function expandWorksContent(props: WorkDetailProps): void {
  props.toggleWorksContentExpand(!props.isExpandWorksContent)
}

const enhance = compose<
  WorkDetailProps & WorkDetailWithHandlers,
  WorkDetailProps
>(
  withHandlers({
    expandWorksContent: (props: WorkDetailProps) => () =>
      expandWorksContent(props),
    hideOverlay: (props: WorkDetailProps) => () => hideOverlay(props),
    onEnteredShowWorksContent: (props: WorkDetailProps) => () =>
      onEnteredShowWorksContent(props),
    showWorksContent: (props: WorkDetailProps) => () => showWorksContent(props),
    onExitedOverlay: (props: WorkDetailProps) => () => onExitedOverlay(props),
  }),
)

export type WorkDetailWithHandlers = {
  [P in
    | 'expandWorksContent'
    | 'hideOverlay'
    | 'onEnteredShowWorksContent'
    | 'showWorksContent'
    | 'onExitedOverlay']: () => any
}

export const WorkDetailComponent = (
  props: WorkDetailProps & WorkDetailWithHandlers,
) => {
  const {
    isExpandWorksContent,
    isShowWorksContent,
    isShowWorksContentAnimation,
    updateWorkContentImg,
    isShowOverlay,
    circleStyle,
    targetWork,
    workContentImg,
  } = props

  const compressIconClass = classnames('fa', {
    'fa-expand': !isExpandWorksContent,
    'fa-compress': isExpandWorksContent,
  })

  const worksBackButtonClass = classnames('fa fa-angle-left icon', {
    expanded: isExpandWorksContent,
  })

  return (
    <div className="work-detail">
      <CSSTransition
        in={isShowWorksContent}
        appear={true}
        classNames="slide-works"
        timeout={1000}
        onEntered={props.onEnteredShowWorksContent}
      >
        <CSSTransition
          in={isExpandWorksContent}
          appear={true}
          classNames="expand-works"
          timeout={1000}
        >
          <div className="work-detail-Overlay">
            <div
              className="work-detail-ExpandOverlay"
              onClick={props.expandWorksContent}
            >
              <i className={compressIconClass} />
            </div>
            <div className="work-detail-Content_Wrapper">
              <Animated
                animationIn="fadeInDown"
                animationOut="fadeOut"
                isVisible={isShowWorksContentAnimation}
              >
                <h2 className="work-detail-Content_Heading">EC Website</h2>
              </Animated>
              <Animated
                animationIn="fadeInDown"
                animationOut="fadeOut"
                animationInDelay={300}
                isVisible={isShowWorksContentAnimation}
              >
                <WorkDetailContent
                  updateWorkContentImg={updateWorkContentImg}
                  targetWork={targetWork}
                />
              </Animated>
            </div>
          </div>
        </CSSTransition>
      </CSSTransition>
      <Animated
        className="work-detail-ContentLeftWrapper"
        animationIn="fadeIn"
        animationOut="fadeOut"
        animateOnMount={false}
        isVisible={isShowWorksContentAnimation}
      >
        <div className="work-detail-ContentLeft">
          <div className="work-detail-ContentLeft_Inner">
            <Animated
              animationIn="fadeInDown"
              animationOut="fadeOut"
              animationInDelay={300}
              isVisible={isShowWorksContentAnimation}
            >
              <div className="work-detail-Uncover">
                <a href="https://google.com" target="blank">
                  <img src={workContentImg} />
                </a>

                <div className="work-detail-Uncover_Slices">
                  {renderWorkContentImgAnimation(props)}
                </div>
              </div>
            </Animated>
          </div>
        </div>
      </Animated>

      <CSSTransition
        in={isShowOverlay}
        appear={true}
        classNames="scale"
        timeout={1000}
        onEnter={props.showWorksContent}
        onExited={props.onExitedOverlay}
      >
        <div className="work-detail-Circle" style={circleStyle} />
      </CSSTransition>

      <CSSTransition
        in={isShowWorksContent}
        appear={true}
        classNames="slide-button"
        timeout={1000}
      >
        <div className="work-detail-BackButton">
          <i className={worksBackButtonClass} onClick={props.hideOverlay} />
        </div>
      </CSSTransition>
    </div>
  )
}

// NOTE: might not want to use lifecycle in recompose
// @see https://github.com/acdlite/recompose/issues/653#issuecomment-383645282
export const Enhanced = enhance(WorkDetailComponent)

export default class WorkDetail extends React.Component<WorkDetailProps, {}> {
  constructor(props: WorkDetailProps) {
    super(props)
  }

  public async componentDidMount(): Promise<void> {
    this.props.toggleOverlay(true)

    let { id } = this.props.match.params
    id = parseInt(id, 10)

    await this.props.fetchWork(id)

    if (!this.props.targetWork) this.props.history.push('/')
  }

  public render() {
    return <Enhanced {...this.props} />
  }
}
