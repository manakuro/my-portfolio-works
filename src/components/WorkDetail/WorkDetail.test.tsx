import * as React from 'react'
import { shallow } from 'enzyme'
import * as WorkDetail from './WorkDetail'
import {
  WorkDetailProps,
  WorkDetailWithHandlers,
} from '@/components/WorkDetail/WorkDetail'

const propsData: WorkDetailProps & WorkDetailWithHandlers = {
  toggleOverlay: jest.fn(),
  toggleWorksContent: jest.fn(),
  toggleWorksContentAnimation: jest.fn(),
  toggleWorksContentExpand: jest.fn(),
  updateCircle: jest.fn(),
  updateWorkContentImg: jest.fn(),
  fetchWork: jest.fn(),

  workContentImg: 'http://sample.jpg',
  isShowOverlay: false,
  isShowWorksContent: false,
  isExpandWorksContent: false,
  isShowWorksContentAnimation: false,

  circleStyle: {
    top: 'calc(50% - 50px)',
    left: 'calc(50% - 50px)',
  },

  targetWork: {
    id: 1,
    title: 'EC Website',
    description: `
      Iâ€™ve been a CMT for ten years now, and I have, literally and
      figuratively, held the pulse of a steaming cross-section of
      San Franciscans
      `,
    img: 'http://sample.jpg',
    component: 'WorkOne',
    languages: [1, 2],
  },
  history: {
    push: jest.fn(),
    goBack: jest.fn(),
    location: {},
  } as any,
  match: {
    params: {
      id: 1,
    },
  } as any,
  expandWorksContent: jest.fn(),
  hideOverlay: jest.fn(),
  onEnteredShowWorksContent: jest.fn(),
  showWorksContent: jest.fn(),
  onExitedOverlay: jest.fn(),
}

describe('WorkDetail', () => {
  let props: WorkDetailProps & WorkDetailWithHandlers
  beforeEach(() => {
    props = { ...propsData }
  })

  it('should render component', () => {
    const wrapper = shallow(<WorkDetail.WorkDetailComponent {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('WorkDetail class', () => {
    let wrapper: any

    it('should render enhanced component', () => {
      wrapper = shallow(<WorkDetail.default {...props} />)
      expect(wrapper).toMatchSnapshot()
    })

    describe('componentDidMount', () => {
      it('should show overlay', async () => {
        wrapper = await shallow(<WorkDetail.default {...props} />)

        expect(props.toggleOverlay).toHaveBeenCalledWith(true)
        expect(props.fetchWork).toHaveBeenCalledWith(1)
      })
      it('should go back to top page', async () => {
        props.targetWork = null

        wrapper = await shallow(<WorkDetail.default {...props} />)

        expect(props.toggleOverlay).toHaveBeenCalledWith(true)
        expect(props.fetchWork).toHaveBeenCalledWith(1)
        expect(props.history.push).toHaveBeenCalledWith('/')
      })
    })
  })

  describe('renderWorkContentImgAnimation', () => {
    it('should render work content image with animation', () => {
      props.isShowWorksContent = true
      props.isExpandWorksContent = false
      const result: any = WorkDetail.renderWorkContentImgAnimation(props)

      expect(shallow(result)).toMatchSnapshot()
    })
    it('should return null', () => {
      props.isShowWorksContent = false
      props.isExpandWorksContent = true
      const result: any = WorkDetail.renderWorkContentImgAnimation(props)

      expect(result).toEqual(null)
    })
  })

  describe('goBack', () => {
    it('should go back', () => {
      props.history.location.state = {
        fromWorkListItem: true,
      }

      WorkDetail.goBack(props)
      expect(props.history.goBack).toHaveBeenCalled()
    })

    it('should go to top when user come from other url', () => {
      props.history.location.state = null

      WorkDetail.goBack(props)
      expect(props.history.push).toHaveBeenCalledWith('/')
    })
  })

  describe('hideOverlay', () => {
    it('should hide work content', () => {
      WorkDetail.hideOverlay(props)

      expect(props.toggleWorksContent).toHaveBeenCalledWith(false)
      expect(props.toggleOverlay).toHaveBeenCalledWith(false)
      expect(props.toggleWorksContentAnimation).toHaveBeenCalledWith(false)
      expect(props.toggleWorksContentExpand).toHaveBeenCalledWith(false)
    })
  })

  describe('onExitedOverlay', () => {
    it('should initialize circle style and go back to home', () => {
      WorkDetail.onExitedOverlay(props)

      expect(props.updateCircle).toHaveBeenCalledWith({
        top: '0px',
        left: '-1000px',
      })
    })
  })

  describe('showWorksContent', () => {
    it('should show work content', () => {
      WorkDetail.showWorksContent(props)

      expect(props.toggleWorksContent).toHaveBeenCalledWith(true)
    })
  })

  describe('onEnteredShowWorksContent', () => {
    it('should show work content animation', () => {
      WorkDetail.onEnteredShowWorksContent(props)

      expect(props.toggleWorksContentAnimation).toHaveBeenCalledWith(true)
    })
  })

  describe('expandWorksContent', () => {
    it('should expand work content', () => {
      WorkDetail.expandWorksContent(props)

      expect(props.toggleWorksContentExpand).toHaveBeenCalledWith(
        !props.isExpandWorksContent,
      )
    })
  })
})
