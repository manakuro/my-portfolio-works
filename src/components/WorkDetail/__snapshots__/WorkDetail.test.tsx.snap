// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkDetail WorkDetail class should render enhanced component 1`] = `
<withHandlers(Component)
  circleStyle={
    Object {
      "left": "calc(50% - 50px)",
      "top": "calc(50% - 50px)",
    }
  }
  expandWorksContent={[MockFunction]}
  fetchWork={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  hideOverlay={[MockFunction]}
  history={
    Object {
      "goBack": [MockFunction],
      "location": Object {},
      "push": [MockFunction],
    }
  }
  isExpandWorksContent={false}
  isShowOverlay={false}
  isShowWorksContent={false}
  isShowWorksContentAnimation={false}
  match={
    Object {
      "params": Object {
        "id": 1,
      },
    }
  }
  onEnteredShowWorksContent={[MockFunction]}
  onExitedOverlay={[MockFunction]}
  showWorksContent={[MockFunction]}
  targetWork={
    Object {
      "component": "WorkOne",
      "description": "
      I’ve been a CMT for ten years now, and I have, literally and
      figuratively, held the pulse of a steaming cross-section of
      San Franciscans
      ",
      "id": 1,
      "img": "http://sample.jpg",
      "languages": Array [
        1,
        2,
      ],
      "title": "EC Website",
    }
  }
  toggleOverlay={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
        ],
      ],
    }
  }
  toggleWorksContent={[MockFunction]}
  toggleWorksContentAnimation={[MockFunction]}
  toggleWorksContentExpand={[MockFunction]}
  updateCircle={[MockFunction]}
  updateWorkContentImg={[MockFunction]}
  workContentImg="http://sample.jpg"
/>
`;

exports[`WorkDetail renderWorkContentImgAnimation should render work content image with animation 1`] = `
<React.Fragment>
  <div
    className="work-detail-Uncover_Slice"
    key="0"
  />
  <div
    className="work-detail-Uncover_Slice"
    key="1"
  />
  <div
    className="work-detail-Uncover_Slice"
    key="2"
  />
  <div
    className="work-detail-Uncover_Slice"
    key="3"
  />
  <div
    className="work-detail-Uncover_Slice"
    key="4"
  />
  <div
    className="work-detail-Uncover_Slice"
    key="5"
  />
</React.Fragment>
`;

exports[`WorkDetail should render component 1`] = `
<div
  className="work-detail"
>
  <CSSTransition
    appear={true}
    classNames="slide-works"
    in={false}
    onEntered={[MockFunction]}
    timeout={1000}
  >
    <CSSTransition
      appear={true}
      classNames="expand-works"
      in={false}
      timeout={1000}
    >
      <div
        className="work-detail-Overlay"
      >
        <div
          className="work-detail-ExpandOverlay"
          onClick={[MockFunction]}
        >
          <i
            className="fa fa-expand"
          />
        </div>
        <div
          className="work-detail-Content_Wrapper"
        >
          <Animated
            animateOnMount={true}
            animationIn="fadeInDown"
            animationInDelay={0}
            animationOut="fadeOut"
            animationOutDelay={0}
            className=""
            isVisible={false}
            style={Object {}}
          >
            <h2
              className="work-detail-Content_Heading"
            >
              EC Website
            </h2>
          </Animated>
          <Animated
            animateOnMount={true}
            animationIn="fadeInDown"
            animationInDelay={300}
            animationOut="fadeOut"
            animationOutDelay={0}
            className=""
            isVisible={false}
            style={Object {}}
          >
            <WorkDetailContent
              targetWork={
                Object {
                  "component": "WorkOne",
                  "description": "
      I’ve been a CMT for ten years now, and I have, literally and
      figuratively, held the pulse of a steaming cross-section of
      San Franciscans
      ",
                  "id": 1,
                  "img": "http://sample.jpg",
                  "languages": Array [
                    1,
                    2,
                  ],
                  "title": "EC Website",
                }
              }
              updateWorkContentImg={[MockFunction]}
            />
          </Animated>
        </div>
      </div>
    </CSSTransition>
  </CSSTransition>
  <Animated
    animateOnMount={false}
    animationIn="fadeIn"
    animationInDelay={0}
    animationOut="fadeOut"
    animationOutDelay={0}
    className="work-detail-ContentLeftWrapper"
    isVisible={false}
    style={Object {}}
  >
    <div
      className="work-detail-ContentLeft"
    >
      <div
        className="work-detail-ContentLeft_Inner"
      >
        <Animated
          animateOnMount={true}
          animationIn="fadeInDown"
          animationInDelay={300}
          animationOut="fadeOut"
          animationOutDelay={0}
          className=""
          isVisible={false}
          style={Object {}}
        >
          <div
            className="work-detail-Uncover"
          >
            <a
              href="https://google.com"
              target="blank"
            >
              <img
                src="http://sample.jpg"
              />
            </a>
            <div
              className="work-detail-Uncover_Slices"
            />
          </div>
        </Animated>
      </div>
    </div>
  </Animated>
  <CSSTransition
    appear={true}
    classNames="scale"
    in={false}
    onEnter={[MockFunction]}
    onExited={[MockFunction]}
    timeout={1000}
  >
    <div
      className="work-detail-Circle"
      style={
        Object {
          "left": "calc(50% - 50px)",
          "top": "calc(50% - 50px)",
        }
      }
    />
  </CSSTransition>
  <CSSTransition
    appear={true}
    classNames="slide-button"
    in={false}
    timeout={1000}
  >
    <div
      className="work-detail-BackButton"
    >
      <i
        className="fa fa-angle-left icon"
        onClick={[MockFunction]}
      />
    </div>
  </CSSTransition>
</div>
`;
